### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by SSEKPIUS**
> Pay special attention to the "Single-responsibility Principle".

Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -16,7 +16,6 @@ def initialize
 
   def list_books
     puts 'No books added!' if @books.empty?
-
     @books.each { |book| puts "Title: #{book.title}, author: #{book.author}" }
     puts ''
   end
```
Comment (danmainah) [inline app.rb:20]: - [ ] @SSEKPIUS and @Tamunokuro kindly ensure all requests for input and input validation should be placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input

Comment (danmainah) [reply app.rb:20]: - [ ] @team also ensure your methods are following `Single responsibility rule` ,eg the create person method should have a statement to create student and teacher, the create student and create teacher method should be called in create person method to pass specialized parameter  unique to teachers and students

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @SSEKPIUS and @Tamunokuro , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @SSEKPIUS and @Tamunokuro,

✔️ Implementation of SOLID principles

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**